src/parsers/power_system_table_data.jl

```diff
@@ -807,99 +807,160 @@ function make_generator(data::PowerSystemTableData, gen, cost_colnames, bus, gen
+        CostCurve(var_cost, UnitSystem.NATURAL_UNITS),
+        gen.fixed_cost,
+        startup_cost,
+        shutdown_cost,
+    )
+    return op_cost
+end
+
+function make_cost(
+    ::Type{T},
+    data,
+    gen,
+    cost_colnames::_HeatRateColumns,
+) where {T <: HydroGen}
+    fuel_price = gen.fuel_price / 1000.0
+    cost_pairs = get_cost_pairs(gen, cost_colnames)
+    var_cost, fixed = create_pwinc_cost(cost_pairs)
+    op_cost = HydroGenerationCost(
+        FuelCurve(var_cost, UnitSystem.NATURAL_UNITS, fuel_price),
+        fixed * fuel_price)
+    return op_cost
+end
+
+function make_cost(
+    ::Type{T},
+    data,
+    gen,
+    cost_colnames::_CostPointColumns,
+) where {T <: HydroGen}
+    cost_pairs = get_cost_pairs(gen, cost_colnames)
+    var_cost = create_pwl_cost(cost_pairs)
+    op_cost = HydroGenerationCost(
+        CostCurve(var_cost, UnitSystem.NATURAL_UNITS),
+        gen.fixed_cost)
+    return op_cost
+end
+
+function make_cost(
+    ::Type{T},
+    data,
+    gen,
+    cost_colnames::_HeatRateColumns,
+) where {T <: RenewableGen}
+    @warn "Heat rate parsing not valid for RenewableGen replacing with zero cost"
+    var_cost = CostCurve(;
+        value_curve = LinearCurve(0.0),
+        power_units = UnitSystem.NATURAL_UNITS,
+        vom_cost = isnothing(gen.variable_cost) ? 0.0 : gen.variable_cost,
+    )
+    op_cost = RenewableGenerationCost(var_cost)
+    return op_cost
+end
+
+function make_cost(
+    ::Type{T},
+    data,
+    gen,
+    cost_colnames::_CostPointColumns,
+) where {T <: RenewableGen}
+    cost_pairs = get_cost_pairs(gen, cost_colnames)
+    var_cost = CostCurve(;
+        value_curve = cost_pairs,
+        power_units = UnitSystem.NATURAL_UNITS,
+        vom_cost = isnothing(gen.variable_cost) ? 0.0 : gen.variable_cost,
+    )
+    op_cost = RenewableGenerationCost(var_cost)
+    return op_cost
+end
+
+function get_cost_pairs(gen::NamedTuple, cost_colnames)
+    base_power = gen.base_mva * gen.active_power_limits_max
+    vals = []
+    for (c, pt) in cost_colnames.columns
+        x = getfield(gen, pt)
+        y = getfield(gen, c)
+
+        if !in(nothing, [x, y])
+            push!(vals,
+                (x = tryparse(Float64, string(x)) * base_power,
+                    y = tryparse(Float64, string(y))))
         end
-        var_cost = PiecewiseLinearData([(x, y) for (y, x) in var_cost])
+    end
+
+    last_increasing_point = findfirst(x -> x < 0.0, [diff(getfield.(vals, :x))..., -Inf])
+    return vals[1:last_increasing_point]
+end
 
-    elseif length(var_cost) == 1
+function create_pwl_cost(
+    cost_pairs,
+)
+    if length(cost_pairs) > 1
+        var_cost = PiecewisePointCurve(PiecewiseLinearData(cost_pairs))
+    elseif length(cost_pairs) == 1
         # if there is only one point, use it to determine the constant $/MW cost
-        var_cost = LinearFunctionData(var_cost[1][1] * fuel_cost + vom)
-        fixed = 0.0
+        var_cost = LinearCurve(cost_pairs[1].y)
```

Comment:
Adding to my suspicion is the fact that we're doing the same thing below in `create_pwinc_cost`, where the `y` values are supposed to have different units

Reply:
The non-fuel option is more flexible since we don't really have data in that format. So, I agree with this point and made the suggested change.

Reply:
Am I understanding correctly that we get to make up the spec here since there's no existing data we're trying to match? The implementation as it now stands seems at least internally consistent, though I wonder why we decided that cost curves should be input-output curves and fuel curves should be incremental. Is there a rationale for that?